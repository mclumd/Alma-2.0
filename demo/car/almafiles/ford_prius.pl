% Example 1: we see a car, and we see it has an empty gas tank
obs(quote(rel(gas_tank_car, ford1))).
obs(quote(not(rel(fueled, ford1)))).

fif(rel(car, X), rel(can_drive, X)).


% Example 2: a charged prius with an empty gas tank
obs(quote(rel(electric_car, prius1))).
obs(quote(rel(gas_tank_car, prius1))).
obs(quote(rel(charged, prius1))).
obs(quote(not(rel(fueled, prius1)))).

fif(rel(powered, X), rel(can_drive, X)).
fif(not(rel(powered, X)), not(rel(can_drive, X))).

fif(and(rel(gas_tank_car, X), rel(fueled, X)), rel(powered, X)).
%fif(rel(empty_gas_tank, X), not(rel(powered, X))).
fif(and(rel(gas_tank_car, X), not(rel(fueled, X))), not(rel(powered, X))).
fif(and(rel(electric_car, X), rel(charged, X)), rel(powered, X)). 

rel(is_a, gas_tank_car, car).
rel(is_a, electric_car, car).
