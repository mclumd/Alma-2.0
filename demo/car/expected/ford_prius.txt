rel(car, ford1)
rel(gas_tank_car, ford1)
~rel(can_drive, ford1)
~rel(fueled, ford1)
abnormal(ford1, car, can_drive)
rel(car, prius1)
rel(gas_tank_car, prius1)
rel(electric_car, prius1)
rel(charged, prius1)
~rel(fueled, prius1)
rel(can_drive, prius1)
rel(powered, prius1)
abnormal(prius1, gas_tank_car, powered)
